src/App.tsx

```diff
@@ -1,61 +1,139 @@
 import './App.scss';
+import React, { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
+import { TodoList } from './components/TodoList';
+
+function findUser(id: number): User | null {
+  return usersFromServer.find(user => user.id === id) || null;
+}
+
+const todoList: Todo[] = todosFromServer.map(todo => (
+  {
+    ...todo,
+    user: findUser(todo.id),
+  }
+));
+
+function calculateId(todos: Todo[]): number {
+  const idArray = todos.map(todo => todo.id).sort((a, b) => b - a);
+
+  return idArray[0] + 1;
+}
 
 export const App = () => {
+  const [todos, setTodos] = useState(todoList);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [userError, setUserError] = useState(false);
+  const [titleError, setTitleError] = useState(false);
+
+  const clear = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
```

Comment:
Consider destructuring the `value` property from the `event.target` object for better readability: `const { value } = event.target;`